### PR TITLE
Add tests for SlackRecord

### PR DESCRIPTION
**Pull request by an1zhegorodov**
- Added tests for SlackRecord
- Fixed broken tests for SlackHandler
- Extra/conext variables are now stringified if they are arrays
- Removed attachment json_encoding
- Some codereview changes based on stof comments

### CODE REVIEW

**Comment by hkdobrev** [comment]
Thanks! Will try to look at this tonight :)

**Comment by Seldaek** [comment]
Great thanks both, I'll delay the release a bit then.

File: src/Monolog/Handler/Slack/SlackRecord.php [hunk 1]
```diff
@@ -221,4 +191,47 @@ public function stringify($fields)
...
+        $this->formatter = $formatter;
+    }
+
+    /**
+     * Generates attachment field
+     *
+     * @param $title
+     * @param $value
+     * @param $short
+     * @return array
```
Comment (an1zhegorodov) [inline src/Monolog/Handler/Slack/SlackRecord.php]: Add types to phpdoc

Comment (hkdobrev) [inline src/Monolog/Handler/Slack/SlackRecord.php]: Done.

**Comment by hkdobrev** [comment]
I'm closing this as I've cherry-picked the commits and fixed conflicts resulted from some changes I haven't pushed. See origin PR for details https://github.com/Seldaek/monolog/pull/846

Thanks again @an1zhegorodov!

**Pull request closed without merging**